polygeist/tools/cgeist/driver.cc

```diff
@@ -840,6 +909,9 @@ static bool splitCommandLineOptions(int argc, char **argv,
       } else if (ref == "-fsycl-is-device") {
         syclKernelsOnly = true;
         MLIRArgs.push_back(argv[i]);
+      } else if (ref.startswith("-O")) {
+        setMLIROptLevel(argv[i]);
+        LinkageArgs.push_back(argv[i]);
```

Comment:
That is an argument which will also apply when emitting binary and there will be some cases, e.g., `-Oz`, which would currently be impossible to set otherwise. It is true that option passing needs some serious work, as there are options we are not applying correctly. I'm currently working on that.

*Note*: It is true we're currently passing `-On` options twice, but I'd say we want to keep this here. What we want is to remove the ad-hoc code passing the options depending on the values of `cgeist`'s `Optn` options.

Reply:
Thanks for the explanation.